src/app/proposals/draft/components/stages/TempCheckForm.tsx

```diff
@@ -0,0 +1,121 @@
+"use client";
+
+import { useState } from "react";
+import { z } from "zod";
+// import Tenant from "@/lib/tenant/tenant";
+import { useForm } from "react-hook-form";
+import { useRouter } from "next/navigation";
+import { zodResolver } from "@hookform/resolvers/zod";
+import FormCard from "../form/FormCard";
+import FormItem from "../form/FormItem";
+import TextInput from "../form/TextInput";
+import { UpdatedButton } from "@/components/Button";
+import { schema as tempCheckSchema } from "../../schemas/tempCheckSchema";
+import { onSubmitAction as tempCheckAction } from "../../actions/createTempCheck";
+import { ProposalDraft } from "@prisma/client";
+import { useAccount } from "wagmi";
+import Image from "next/image";
+
+const TempCheckForm = ({ draftProposal }: { draftProposal: ProposalDraft }) => {
+  const router = useRouter();
+  const { address } = useAccount();
+  const [isSkipPending, setIsSkipPending] = useState(false);
+  const [isSubmitPending, setIsSubmitPending] = useState(false);
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+  } = useForm<z.output<typeof tempCheckSchema>>({
+    resolver: zodResolver(tempCheckSchema),
+    defaultValues: {
+      temp_check_link: draftProposal.temp_check_link,
+    },
+  });
+
+  const onSubmitSkip = async (data: z.output<typeof tempCheckSchema>) => {
+    setIsSkipPending(true);
+    await sharedOnSubmit(data);
+  };
+
+  const onSubmit = async (data: z.output<typeof tempCheckSchema>) => {
+    setIsSubmitPending(true);
+    await sharedOnSubmit(data);
+  };
+
+  const sharedOnSubmit = async (data: z.output<typeof tempCheckSchema>) => {
+    try {
+      if (!address) {
+        throw new Error("No address connected");
+      }
+
+      const res = await tempCheckAction({
+        ...data,
+        draftProposalId: draftProposal.id,
+        creatorAddress: address,
+      });
+      router.push(`/proposals/draft/${draftProposal.id}?stage=1`);
+    } catch (e) {
+      console.error(e);
+    }
+  };
+
+  return (
+    <form
+    //   action={async (formData: FormData) => {
+    //     await tempCheckAction(formData);
+    //     // this should probably be order in the tenent list + 1
+    //     redirect(`/proposals/draft?stage=1`);
+    //   }}
+    >
+      <FormCard>
+        <FormCard.Section>
+          <div className="w-full rounded-md h-[350px] block relative">
+            <Image
+              src="/images/temp_check.png"
```

Comment:
This needs to be a tenant-specific image... ens_temp_check or something like this. Just a note, no action right now.

Reply:
rename